homeassistant/components/switchbee/climate.py

```diff
@@ -0,0 +1,206 @@
+    ThermostatMode.HEAT: HVACAction.HEATING,
+    ThermostatMode.FAN: HVACAction.FAN,
+}
+
+HVAC_UNIT_SB_TO_HASS = {
+    ThermostatTemperatureUnit.CELSIUS: TEMP_CELSIUS,
+    ThermostatTemperatureUnit.FAHRENHEIT: TEMP_FAHRENHEIT,
+}
+
+SUPPORTED_FAN_MODES = [FAN_AUTO, FAN_HIGH, FAN_MEDIUM, FAN_LOW]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Switchbee thermostat."""
+    coordinator: SwitchBeeCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        SwitchBeeClimate(switchbee_device, coordinator)
+        for switchbee_device in coordinator.data.values()
+        if isinstance(switchbee_device, SwitchBeeThermostat)
+    )
+
+
+class SwitchBeeClimate(SwitchBeeDeviceEntity[SwitchBeeThermostat], ClimateEntity):
+    """Representation of an Switchbee button."""
+
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+    )
+    _attr_fan_modes = SUPPORTED_FAN_MODES
+    _attr_target_temperature_step = 1
+
+    def __init__(
+        self,
+        device: SwitchBeeThermostat,
+        coordinator: SwitchBeeCoordinator,
+    ) -> None:
+        """Initialize the Switchbee switch."""
+        super().__init__(device, coordinator)
+        # set HVAC capabilities
+        self._attr_max_temp = device.max_temperature
+        self._attr_min_temp = device.min_temperature
+        self._attr_temperature_unit = HVAC_UNIT_SB_TO_HASS[device.unit]
+        self._attr_hvac_modes = [HVAC_MODE_SB_TO_HASS[mode] for mode in device.modes]
+        self._attr_hvac_modes.append(HVACMode.OFF)
+        self._update_device_attrs(self._device)
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool mode."""
+        return self._attr_hvac_mode
+
+    @property
+    def fan_mode(self) -> str:
+        """Return the fan setting.
+
+        Requires ClimateEntityFeature.FAN_MODE.
+        """
+        return self._attr_fan_mode
+
+    @property
+    def target_temperature(self) -> int:
+        """Return the temperature we try to reach."""
+        return self._attr_target_temperature
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_device_attrs(
+            cast(SwitchBeeThermostat, self.coordinator.data[self._device.id])
+        )
+        super()._handle_coordinator_update()
+
+    def _update_device_attrs(self, device: SwitchBeeThermostat) -> None:
+        self._attr_hvac_mode: HVACMode = (
+            HVACMode.OFF
+            if device.state == ApiStateCommand.OFF
+            else HVAC_MODE_SB_TO_HASS[device.mode]
+        )
+        self._attr_fan_mode: str = FAN_SB_TO_HASS[device.fan]
+        self._attr_current_temperature: float = device.temperature
+        self._attr_target_temperature: int = device.target_temperature
```

Comment:
```suggestion
        self._attr_fan_mode = FAN_SB_TO_HASS[device.fan]
        self._attr_current_temperature = device.temperature
        self._attr_target_temperature = device.target_temperature
```
They are typed in the base class